frame/bounties/src/lib.rs

```diff
@@ -653,12 +777,613 @@ decl_module! {
+		///
+		/// Beneficiary is paid out with with agreed bounty value.
+		/// SubCurator fee is paid & curator deposit is unreserved.
+		///
+		/// Subbounty must be in PendingPayout state, for
+		/// processing the call. and instance of subbounty is
+		/// deallocated from DB on successful call completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		#[weight = <T as Config>::WeightInfo::claim_subbounty()]
+		fn claim_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+		) {
+			let _ = ensure_signed(origin)?;
+
+			// TODO :: Have to recheck
+			// ignoring the requirement of parent bounty should be active
+			// for claiming the subbounty.
+			// Since subbounty is executed & in waiting period of PendingPayout.
+			// We can gracefully execute this call, without having dependency on
+			// state of parent bounty, This enables to call close_subbounty()
+			// recursively from close_bounty() without any issue.
+			// // Ensure parent bounty is Active
+			// let master_curator = Self::ensure_bounty_active(bounty_id)?;
+
+			// Ensure subbounty is in expected state
+			SubBounties::<T>::try_mutate_exists(
+				bounty_id,
+				subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+					let subbounty = maybe_subbounty
+						.as_mut()
+						.ok_or(Error::<T>::InvalidIndex)?;
+
+					if let SubBountyStatus::PendingPayout {
+						ref subcurator, ref beneficiary, ref unlock_at
+					} = subbounty.status {
+						// Ensure block number is elapsed for
+						// processing the claim.
+						ensure!(
+							system::Module::<T>::block_number() >= *unlock_at,
+							Error::<T>::Premature,
+						);
+
+						// Make curator fee payment
+						let subbounty_account = Self::bounty_account_id(subbounty_id);
+						let balance = T::Currency::free_balance(&subbounty_account);
+						let fee = subbounty.fee.min(balance); // just to be safe
+						let payout = balance.saturating_sub(fee);
+
+						// unreserve the subcurator deposit
+						let _ = T::Currency::unreserve(
+							&subcurator,
+							subbounty.curator_deposit,
+						); // should not fail
+
+						// Make payout to subcurator
+						let _ = T::Currency::transfer(
+							&subbounty_account,
+							&subcurator,
+							fee,
+							AllowDeath,
+						); // should not fail
+
+						// Make payout to beneficiary
+						let _ = T::Currency::transfer(
+							&subbounty_account,
+							beneficiary,
+							payout,
+							AllowDeath,
+						); // should not fail
+
+						// Trigger the SubBountyClaimed event
+						Self::deposit_event(
+							Event::<T>::SubBountyClaimed(
+								bounty_id,
+								subbounty_id,
+								payout,
+								beneficiary.clone(),
+							)
+						);
+
+						// Remove the subbounty from bounty active subbouty list
+						Bounties::<T>::mutate_exists(
+							bounty_id,
+							|maybe_bounty| -> DispatchResult {
```

Comment:
```suggestion
							|maybe_bounty| {
```

it shouldn't return anything